### ISSUE TITLE
Automate WormSim deployment via jenkins script

### ISSUE THREAD

**Issue by gidili** [openworm/org.wormsim.frontend#53]

**Comment by MichaelCurrie** [comment]
Since we use Travis CI for continuous integration and testing, wouldn't it be better to use [their tools](http://docs.travis-ci.com/user/deployment/) for deployment instead of jenkins?  Just asking.

**Comment by gidili** [comment]
@MichaelCurrie I agree that if we manage to get the integrated solution to work all on the same platform it would be a better option. Travis [supports a number of providers](http://docs.travis-ci.com/user/deployment/) and notably has integration with amazon instances via AWS Code deploy but it is somewhat limited, or at least hard to get to work - @tarelli tried but says it assumes you are deploying one repo and not many like we have (one per each geppetto bundle). Jenkins is much more flexible in the sense that you can just list your deployment steps with no limitations on what you can cannot do in terms of commands.

**Comment by slarson** [comment]
Started this process today.  Using [this document](https://docs.google.com/document/d/11G89BpnZLu_3ibzn0EZU_wR5XD9rw0m0crg8I5rv5u8/edit#) as a guide.  Committed [a branch within initial contents to make this happen](https://github.com/openworm/org.wormsim.frontend/tree/docker/docker).

Next steps:

- [ ] Incorporate [the deploy scripts](https://github.com/openworm/org.wormsim.frontend/tree/docker/utilities) in the build.
- [ ] make sure to modify the paths in the scripts (e.g. [this one](https://github.com/openworm/org.wormsim.frontend/blob/master/utilities/deployWormSimData.py#L34))
- [ ] Do a docker build on the docker file.  Likely to fail.  Check in on the logs to see what is not working.  Review the document above to see what is not working
- [ ] Log in to the instance and look around in the directories to see what may be misconfigured
- [ ] Keep iterating the dockerfile until it comes up.

**Comment by VahidGh** [comment]
Hi,

Is it OK to start over the wormsim project with this issue?

**Comment by pgleeson** [comment]
Sure. Not sure what state it is in or how easy it would be to get going inside docker...

**Comment by VahidGh** [comment]
@Padraig,

I've been playing with the docker files for some hours for the past few days.
I could run some through Github actions but the main problem is that the latest version of Wormsim (here) is outdated and maybe incompatible with the current stable versions of other modules in Geppetto.

I'm mainly aiming to use an online version of the Wormsim for educational purposes.
Will try to see if I can find a temporary workaround until we decide if we are going to put some effort for an upgraded version of the Wormsim.

The docker branch can be a good starting point from an integrated development perspective (where all the installation and dependency requirements are gathered in one place), although, from a deployment point of view has also some advantages (security, cloud-friendly, etc...).

Just not sure to wish the worm to find its way out of the docker through a hole or not ;) :D

**Comment by VahidGh** [comment]
**Issue Update: Troubleshooting WormSim and Geppetto Compatibility**

1. **Initial Issue with log4j**  
   - I began by testing the current Docker file and configuration, discovering that log4j is no longer supported. I attempted workarounds by running `mvn install` with options to bypass tests (`-Dmaven.test.skip`, `-DskipTests`, and `-Dmaven.test.skip.exec`) to skip log4j, which is only required for testing. Unfortunately, this did not resolve the error.  
   - It's possible that additional modules beyond `org.geppetto` rely on log4j. As `mvn install` only directly installs `org.geppetto` (which then installs other modules hierarchically), it may require modifying the installation approach for v0.2.6.  
   - [Build result via GitHub Actions](https://github.com/VahidGh/org.wormsim.frontend/actions/runs/11416848011/job/31768507345)

2. **Building the Latest Geppetto Stack Independently (Excluding WormSim)**  
   - I successfully built the latest Geppetto stack (development branch) by independently installing each module without WormSim.
   - [Successful build via GitHub Actions](https://github.com/VahidGh/org.geppetto/actions/runs/11382957545/job/31667579139)

3. **Independent Installations with WormSim v0.2.6 and Geppetto (Current Stack)**  
   - Using the latest installation method with WormSim v0.2.6 and the same Geppetto stack version, I resolved the log4j issue. However, I encountered a new error with a node module required during installation via `grunt`.
   - [Error encountered with node module](https://github.com/VahidGh/org.geppetto/actions/runs/11381588768/job/31663309332)

4. **Independent Installations with WormSim v0.2.6 and Geppetto (Development Stack)**  
   - I then attempted installations using WormSim v0.2.6 alongside the Geppetto development version. This build stalled on a missing `bower` installation due to the outdated "metacell/java-virgo-maven" base container, which prevented installing additional packages within the container.
   - After applying a workaround to resolve the `bower` issue and overcoming further minor issues, I encountered an incompatibility between Geppetto model architectures for the development version and v0.2.6.  
   - [Incompatibility error with development version](https://github.com/VahidGh/org.geppetto/actions/runs/11388543545/job/31686216795)
